### ISSUE TITLE
Problem with removing expression for Locked Position Property

### ISSUE THREAD

**Issue by reneandritsch** [Rainbox-dev/DuAEF_Duik#313]
# Environement
### Tool & Version
- Duik 16.1.0-Alpha-2

### After Effects
- CC2018 (15.1.2)
- English

### OS
- Mac OS 10.13.6

# What's the issue?
If a user removes the expression but by any chance has changed the position to a value that moves the object off the screen while it was locked it will be moved after removing the expression.

![lock](https://user-images.githubusercontent.com/38455050/44870035-53247880-ac8f-11e8-88b1-8f147543ddd1.gif)

Nicholas, what do you think of adding a Layer Marker with for instance with the text ”Locked by Expression” to tell users that there is a lock expression in place? Because in the course of a project you might forget that you locked the layer and if it is not twirled open there is no hint about that.

**Comment by Nico-Duduf** [comment]
Yes #296 might be a solution, but it implies the user always use Duik to remove expressions instead of removing them with Ae...

You're right, it might be good to show that properties are locked. The problem with markers is how can we show which property is locked?
We could show that just "some properties" are locked, we would just have to find how to do this nicely when there already is a Duik marker like "Structure" and others...

Or maybe something else than a marker? I don't know. The other issue is if the property(ies) is(are) unlocked, the way it is shown has to be removed too, which the user may (will probably) not do.

**Comment by reneandritsch** [comment]
A Marker should be sufficient because you just hit “EE” and see all the locked properties. Maybe the Marker says “Hit EE to reveal locked Properties”.

**Comment by reneandritsch** [comment]
In case there is already a marker just add a second one? That might be a rare case anyway.

**Comment by reneandritsch** [comment]
Another idea for marking the layer could be a lock symbol (like one of the controller layers) on top of the center of the layer … hidden with the shy switch. But maybe it‘s not a good idea to put an extra element into the layer panel.

**Comment by Nico-Duduf** [comment]
What about this kind of thing?
![image](https://user-images.githubusercontent.com/8526542/45049458-f7873000-b07e-11e8-9c64-8a93bc1f250c.png)

**Comment by Nico-Duduf** [comment]
Adding a ◘character in the marker

**Comment by Nico-Duduf** [comment]
In this case we could add the marker with only this special character.
I can even store the list of locked properties, hidden in the marker, to 
be able to unlock them in a single click in Duik, and even re-lock them 
later! Thinking about this, there are a lot of things which could be 
done using the fact that markers can store hidden data...

Le 04/09/2018 à 20:22, reneandritsch a écrit :

**Comment by reneandritsch** [comment]
That sounds promising. Looking forward to what you will do with it! Thanks for all your efforts in improving DUIK. It is really an amazing software!
